### PR TITLE
Remove deprecated modules

### LINKED ISSUE #62: Remove deprecated modules and functions
### Description

These modules were moved and marked deprecated. As a part of maintainance and library refresh, we can work on removing placeholders and documentation references

```
grep 'Deprecation'  -nR {tests,pytensor}/**/*.py
tests/tensor/nnet/test_conv.py:88:            with pytest.warns(DeprecationWarning):
tests/tensor/nnet/test_conv.py:602:                        with pytest.warns(DeprecationWarning):
tests/tensor/nnet/test_conv.py:640:        with pytest.warns(DeprecationWarning):
tests/tensor/nnet/test_conv.py:653:        with pytest.warns(DeprecationWarning):
tests/tensor/nnet/test_conv.py:666:        with pytest.warns(DeprecationWarning):
tests/tensor/nnet/test_conv.py:679:        with pytest.warns(DeprecationWarning):
tests/tensor/nnet/test_conv.py:692:        with pytest.warns(DeprecationWarning):
tests/tensor/nnet/test_conv.py:705:        with pytest.warns(DeprecationWarning):
tests/tensor/nnet/test_conv.py:718:        with pytest.warns(DeprecationWarning):
tests/tensor/nnet/test_conv.py:731:        with pytest.warns(DeprecationWarning):
tests/tensor/nnet/test_conv.py:744:        with pytest.warns(DeprecationWarning):
tests/tensor/nnet/test_conv.py:757:        with pytest.warns(DeprecationWarning):
tests/tensor/test_basic.py:1336:        with pytest.warns(DeprecationWarning):
tests/tensor/test_math_scipy.py:729:    with pytest.warns(DeprecationWarning):
tests/tensor/test_subtensor.py:1205:                        # with NumPy 1.10 will raise a Deprecation warning.
tests/tensor/test_type.py:333:    with pytest.warns(DeprecationWarning, match=".*broadcastable.*"):
tests/tensor/test_type.py:338:    with pytest.warns(DeprecationWarning, match=".*broadcastable.*"):
tests/tensor/test_type.py:359:        DeprecationWarning, match="The `broadcastable` keyword is deprecated"
tests/test_config.py:29:    with pytest.warns(DeprecationWarning, match="instead"):
tests/test_config.py:35:    with pytest.warns(DeprecationWarning, match="instead"):
tests/test_config.py:52:    with pytest.warns(DeprecationWarning):
tests/test_config.py:63:    with pytest.warns(DeprecationWarning):
pytensor/assert_op.py:7:    DeprecationWarning,
pytensor/configparser.py:84:            DeprecationWarning,
pytensor/configparser.py:193:        # The following code adds redirects that spill DeprecationWarnings
pytensor/configparser.py:572:    config settings, but raises DeprecationWarnings with instructions to use `pytensor.config`.
pytensor/configparser.py:583:            DeprecationWarning,
pytensor/configparser.py:593:            DeprecationWarning,
pytensor/configparser.py:637:            warn(msg, DeprecationWarning, stacklevel=2)
pytensor/gradient.py:2142:        category=DeprecationWarning,
pytensor/gradient.py:2376:            warn(msg, DeprecationWarning, stacklevel=2)
pytensor/graph/kanren.py:6:    DeprecationWarning,
pytensor/graph/optdb.py:6:    DeprecationWarning,
pytensor/graph/optdb.py:26:            warn(msg, DeprecationWarning, stacklevel=2)
pytensor/graph/opt.py:6:    DeprecationWarning,
pytensor/graph/opt.py:26:            warn(msg, DeprecationWarning, stacklevel=2)
pytensor/graph/opt_utils.py:6:    DeprecationWarning,
pytensor/graph/opt_utils.py:26:            warn(msg, DeprecationWarning, stacklevel=2)
pytensor/graph/rewriting/basic.py:116:            DeprecationWarning,
pytensor/graph/rewriting/basic.py:2312:            DeprecationWarning,
pytensor/graph/rewriting/basic.py:3261:            warn(msg, DeprecationWarning, stacklevel=2)
pytensor/graph/rewriting/db.py:577:            warn(msg, DeprecationWarning, stacklevel=2)
pytensor/graph/rewriting/utils.py:68:            DeprecationWarning,
pytensor/graph/rewriting/utils.py:272:            warn(msg, DeprecationWarning, stacklevel=2)
pytensor/graph/toolbox.py:7:    DeprecationWarning,
pytensor/graph/unify.py:6:    DeprecationWarning,
pytensor/__init__.py:196:            warn(msg, DeprecationWarning, stacklevel=2)
pytensor/link/jax/jax_dispatch.py:7:    DeprecationWarning,
pytensor/link/jax/jax_linker.py:7:    DeprecationWarning,
pytensor/printing.py:199:            DeprecationWarning,
pytensor/sandbox/rng_mrg.py:43:    DeprecationWarning,
pytensor/sandbox/rng_mrg.py:709:                DeprecationWarning,
pytensor/sandbox/rng_mrg.py:1121:            DeprecationWarning,
pytensor/scalar/basic.py:4471:            warn(msg, DeprecationWarning, stacklevel=2)
pytensor/scalar/basic_scipy.py:7:    DeprecationWarning,
pytensor/scan/basic.py:507:                    raise DeprecationWarning(
pytensor/scan/opt.py:6:    DeprecationWarning,
pytensor/sparse/opt.py:6:    DeprecationWarning,
pytensor/tensor/basic_opt.py:6:    DeprecationWarning,
pytensor/tensor/basic.py:2596:            DeprecationWarning,
pytensor/tensor/math_opt.py:6:    DeprecationWarning,
pytensor/tensor/math.py:3153:            warn(msg, DeprecationWarning, stacklevel=2)
pytensor/tensor/nnet/basic.py:2168:            warn(msg, DeprecationWarning, stacklevel=2)
pytensor/tensor/nnet/conv.py:112:        DeprecationWarning,
pytensor/tensor/nnet/conv.py:410:            DeprecationWarning,
pytensor/tensor/nnet/__init__.py:7:    DeprecationWarning,
pytensor/tensor/nnet/opt.py:6:    DeprecationWarning,
pytensor/tensor/opt_uncanonicalize.py:6:    DeprecationWarning,
pytensor/tensor/random/opt.py:6:    DeprecationWarning,
pytensor/tensor/rewriting/basic.py:1295:        warn(msg, DeprecationWarning, stacklevel=2)
pytensor/tensor/sharedvar.py:71:            DeprecationWarning,
pytensor/tensor/signal/conv.py:17:    DeprecationWarning,
pytensor/tensor/signal/pool.py:25:    DeprecationWarning,
pytensor/tensor/signal/pool.py:111:                category=DeprecationWarning,
pytensor/tensor/signal/pool.py:129:                category=DeprecationWarning,
pytensor/tensor/signal/pool.py:145:                category=DeprecationWarning,
pytensor/tensor/signal/pool.py:158:            category=DeprecationWarning,
pytensor/tensor/signal/pool.py:227:                category=DeprecationWarning,
pytensor/tensor/signal/pool.py:245:                category=DeprecationWarning,
pytensor/tensor/signal/pool.py:261:                category=DeprecationWarning,
pytensor/tensor/signal/pool.py:274:            category=DeprecationWarning,
pytensor/tensor/signal/pool.py:391:                    category=DeprecationWarning,
pytensor/tensor/signal/pool.py:409:                    category=DeprecationWarning,
pytensor/tensor/signal/pool.py:426:                    category=DeprecationWarning,
pytensor/tensor/signal/pool.py:1040:                    category=DeprecationWarning,
pytensor/tensor/signal/pool.py:1058:                    category=DeprecationWarning,
pytensor/tensor/signal/pool.py:1074:                    category=DeprecationWarning,
pytensor/tensor/slinalg.py:868:            warn(msg, DeprecationWarning, stacklevel=2)
pytensor/tensor/subtensor_opt.py:6:    DeprecationWarning,
pytensor/tensor/type.py:98:                DeprecationWarning,
pytensor/tensor/type.py:127:                DeprecationWarning,
pytensor/utils.py:153:    when the function is used first time and filter is set for show DeprecationWarning.
pytensor/utils.py:173:                    category=DeprecationWarning,
```

### LINKED ISSUE #110: pytensor.graph.sched is not referenced anywhere
### Description

Seems like `pytensor.graph.sched` only appears in tests where it is tested and in `pytensor.tensor.io` where a utility small function is imported from the module. Should we get rid of this module?

https://github.com/search?q=repo%3Apymc-devs%2Fpytensor+graph.sched+language%3APython&type=code&l=Python

### LINKED ISSUE #120: Remove MRG_RandomStream?
### Description

there are random modules that we deprecated in pymc, not sure if we ever want them here
https://github.com/pymc-devs/pytensor/blob/main/pytensor/sandbox/rng_mrg.py
https://github.com/pymc-devs/pytensor/blob/main/pytensor/sandbox/multinomial.py

### LINKED ISSUE #124: Remove deprecated pytensor.tensor.nnet,signal
### Description

```
python setup.py version
running version
keywords are unexpanded, not using
got version from VCS {'version': '2.8.10+90.g10610f5bb', 'full-revisionid': '10610f5bbe8396cf51791f9c1c46884c75d1df57', 'dirty': False, 'error': None, 'date': '2022-12-13T23:53:11+0300'}
Version: 2.8.10+90.g10610f5bb
 full-revisionid: 10610f5bbe8396cf51791f9c1c46884c75d1df57
 dirty: False
 date: 2022-12-13T23:53:11+0300
```
I guess we consider to bump version very soon. This module was marked for removal in the nest release.

https://github.com/pymc-devs/pytensor/blob/21d723bd44749f540f86db451f6280753289c30d/pytensor/tensor/nnet/__init__.py#L4-L9

https://github.com/pymc-devs/pytensor/blob/21d723bd44749f540f86db451f6280753289c30d/pytensor/tensor/signal/conv.py#L14-L19

https://github.com/pymc-devs/pytensor/blob/21d723bd44749f540f86db451f6280753289c30d/pytensor/tensor/signal/pool.py#L22-L27

### LINKED ISSUE #125: Consider removing deprecated config api
THe deprecation was introduced by @michaelosthege 
https://github.com/pymc-devs/pytensor/blob/21d723bd44749f540f86db451f6280753289c30d/pytensor/configparser.py#L68-L86

### LINKED ISSUE #128: Do we still need to support Sympy?
### Description

https://github.com/pymc-devs/pytensor/blob/43d91d0a6efdf8f86abf5b6198ebaf074eff586c/pytensor/scalar/basic_sympy.py#L29

Tests pass so far

image

### PR DESCRIPTION
**Pull request by ferrine**
### Motivation for these changes
These modules/functions were not used anymore or moved to a different place long time ago

* closes #62 
* closes #110 
* closes #120 
* closes #124
* closes #125
* closes #128

### Checklist
+ [x] Explain motivation and implementation 👆
+ [x] Make sure that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html).
+ [x] Link relevant issues, preferably in [nice commit messages](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).
+ [x] The commits correspond to [_relevant logical changes_](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes). Note that if they don't, we will [rewrite/rebase/squash the git history](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History#_rewriting_history) before merging.
+ [x] Are the changes covered by tests and docstrings?
+ [x] Fill out the short summary sections 👇

## Major / Breaking Changes
- None

## Bugfixes / New features
- None

## Docs / Maintenance
- Deprecated functionality is removed

### CODE REVIEW

**Review by twiecki — APPROVED** [review]

**Comment by ferrine** [comment]
hits the #119 issue

**Comment by ferrine** [comment]
@ricardoV94 I want to make this PR fat, that will get rid of everything we do not want to support, I'll continue opening issues to validate we indeed want to remove some part of the code. Does it resonate with you? Opening small individual PRs is overwhelming

**Comment by ferrine** [comment]
also close #120

File: pytensor/utils.py [hunk 1]
```diff
@@ -111,6 +111,17 @@ def cmp(x, y):
     return (x > y) - (x < y)
 
 
+def key_to_cmp(key):
```
Comment (ricardoV94) [inline pytensor/utils.py:114]: Was this used anywhere else? I saw it was introduced in the same PR as the MIP

**Comment by ricardoV94** [comment]
> @ricardoV94 I want to make this PR fat, that will get rid of everything we do not want to support, I'll continue opening issues to validate we indeed want to remove some part of the code. Does it resonate with you? Opening small individual PRs is overwhelming

Sure. I am going to "request changes", because you mention this PR closes #120 but it doesn't yet.

**Review by ricardoV94 — CHANGES_REQUESTED** [review]
It doesn't yet close #120

**Comment by ferrine** [comment]
> > @ricardoV94 I want to make this PR fat, that will get rid of everything we do not want to support, I'll continue opening issues to validate we indeed want to remove some part of the code. Does it resonate with you? Opening small individual PRs is overwhelming
> 
> Sure. I am going to "request changes", because you mention this PR closes #120 but it doesn't yet.

I open issues when I discover something new, so there will me more referenced issues

File: pytensor/utils.py (hunk 1 above)
Comment (ferrine) [reply pytensor/utils.py:114]: This function was used only once for the sched module

Comment (ricardoV94) [reply pytensor/utils.py:114]: Can we remove it then?

Comment (ferrine) [reply pytensor/utils.py]: just did

**Comment by twiecki** [comment]
Are we keeping `conv2d`?

**Comment by ferrine** [comment]
> Are we keeping `conv2d`?

We do, we even continue testing that

**Review by michaelosthege — COMMENTED** [review]
Wow, so many deletions! That's awesome.

**Comment by ferrine** [comment]
I've modifies mypy script to reflect the updated fileset as suggested in #77 by @michaelosthege

**Comment by ricardoV94** [comment]
Now this is some house cleaning alright!

image

**Comment by ferrine** [comment]
Who is going to press the nuke button?

File: doc/library/tensor/basic.rst [hunk 2]
```diff
@@ -620,7 +621,7 @@ dimensions, see :meth:`_tensor_py_operators.dimshuffle`.
 
     Reshape `x` by left padding the shape with `n_ones` 1s. Note that all
     this new dimension will be broadcastable. To make them non-broadcastable
-    see the :func:`unbroadcast`.
+    see the :func:`specify_shape`.
```
Comment (ricardoV94) [inline doc/library/tensor/basic.rst:624]: This is wrong. To make something non-broadcastable you have to use unbroadcsat

**Review by ricardoV94 — CHANGES_REQUESTED** [review]
Everything else seems good for me, just an error in the docs that were changed

**Review by ricardoV94 — APPROVED** [review]

**Comment by ferrine** [comment]
nuke

**Comment by twiecki** [comment]
Epic.

**Pull request merged**